defopt.py

```diff
@@ -208,6 +218,9 @@ def _populate_parser(func, parser, parsers, short):
             continue
         if positional:
             kwargs['_positional'] = True
+            if default != param.empty:
```

Comment:
This should be `if hasdefault` since default has already been converted to `SUPPRESS`. The current code causes all positional arguments to be marked as optional. Please include a failing test case for this before fixing.